File: pset6/caesar.py
import cs50
import sys

def main():
    if len(sys.argv) == 2:
        k = int(sys.argv[1])
        print("plaintext: ",end ="")
        input = cs50.get_string();
        
        if input != None:
            print("ciphertext: ",end ="")
            #Traverse Over and Check
            for i in range(len(input)):
                
                c = ord(input[i])
                if c >= (65) and c <= (65+26):
                    c = (c-65+k) % 26
                    print(chr(c+65),end="")
                elif c >= (97)  and c <= (97+26):
                    c = (c-97+k) % 26
                    print(chr(c+97),end="")
                else:
                    print(chr(c),end="")
        
            print()
            exit(0)
        
        else:
            exit(1)
                    
    else:
        print("Usage: ./caesar k")
        exit(1)
        
    exit(0)
if __name__ == "__main__":
    main()
            

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>

int main(int argc,string argv[]){
    if(argc == 2){
        // String to Integer
        int k =  atoi(argv[1]);
        printf("plaintext: ");
        string input = get_string();
        
        // Check if Null or not
        if (input != NULL){
            printf("ciphertext: ");
            // Traverse Over and Check
            for(int i = 0,n = strlen(input); i < n; i++){
                char c = input[i];
                if(c >= 'A' && c <= 'Z'){
                    c = (c-65+k) % 26;
                    printf("%c",(c+65));
                }else if (c >= 'a' && c <= 'z' ){
                    c = (c-97+k) % 26;
                    printf("%c",(c+97));
                }else{
                    printf("%c",c);
                }
            }
        printf("\n");
        return 0;
        }
        else{
            return 1;
        }
    }
    else{
        printf("Usage: ./caesar k\n");
        return 1;
    }
    
    return 0;
}